<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MainView</class>
 <widget class="QWidget" name="MainView">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>935</width>
    <height>622</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>Form</string>
  </property>
  <layout class="QHBoxLayout" name="horizontalLayout_2">
   <item>
    <layout class="QVBoxLayout" name="verticalLayout">
     <item>
      <widget class="ReadMNIST" name="readMNIST" native="true">
       <property name="minimumSize">
        <size>
         <width>157</width>
         <height>228</height>
        </size>
       </property>
      </widget>
     </item>
     <item>
      <widget class="QLabel" name="predictionLbl">
       <property name="styleSheet">
        <string notr="true">border: 1px solid rgba(5, 255, 5, 0.7);
border-radius: 5px;
background-color: rgba(113, 222, 78, 100);
margin: 2px;
padding: 3px;</string>
       </property>
       <property name="text">
        <string>-</string>
       </property>
      </widget>
     </item>
     <item>
      <widget class="QLabel" name="accuracyLbl">
       <property name="styleSheet">
        <string notr="true">border: 1px solid rgba(5, 5, 5, 0.7);
border-radius: 5px;
background-color: rgb(255, 255, 127);
margin: 3px;
padding: 2px;
</string>
       </property>
       <property name="text">
        <string>-</string>
       </property>
      </widget>
     </item>
     <item>
      <widget class="QFrame" name="frame_2">
       <property name="frameShape">
        <enum>QFrame::StyledPanel</enum>
       </property>
       <property name="frameShadow">
        <enum>QFrame::Raised</enum>
       </property>
       <layout class="QVBoxLayout" name="verticalLayout_3">
        <item>
         <widget class="QProgressBar" name="prBar1">
          <property name="sizePolicy">
           <sizepolicy hsizetype="Preferred" vsizetype="Fixed">
            <horstretch>0</horstretch>
            <verstretch>0</verstretch>
           </sizepolicy>
          </property>
          <property name="value">
           <number>0</number>
          </property>
         </widget>
        </item>
       </layout>
      </widget>
     </item>
     <item>
      <widget class="QFrame" name="frame">
       <property name="minimumSize">
        <size>
         <width>0</width>
         <height>140</height>
        </size>
       </property>
       <property name="frameShape">
        <enum>QFrame::StyledPanel</enum>
       </property>
       <property name="frameShadow">
        <enum>QFrame::Raised</enum>
       </property>
       <layout class="QVBoxLayout" name="verticalLayout_2">
        <item>
         <widget class="QPushButton" name="saveBtn">
          <property name="text">
           <string>Save conf</string>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QPushButton" name="loadBtn">
          <property name="text">
           <string>Load conf</string>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QPushButton" name="inputBtn">
          <property name="text">
           <string>Set on input</string>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QPushButton" name="rmBtn">
          <property name="text">
           <string>Remove last layer</string>
          </property>
         </widget>
        </item>
        <item>
         <layout class="QHBoxLayout" name="horizontalLayout">
          <item>
           <widget class="QPushButton" name="pushButton">
            <property name="sizePolicy">
             <sizepolicy hsizetype="Preferred" vsizetype="Fixed">
              <horstretch>3</horstretch>
              <verstretch>0</verstretch>
             </sizepolicy>
            </property>
            <property name="text">
             <string>Add Layer</string>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QSpinBox" name="spinBox">
            <property name="sizePolicy">
             <sizepolicy hsizetype="Preferred" vsizetype="Fixed">
              <horstretch>1</horstretch>
              <verstretch>0</verstretch>
             </sizepolicy>
            </property>
            <property name="minimum">
             <number>1</number>
            </property>
            <property name="maximum">
             <number>9999</number>
            </property>
           </widget>
          </item>
         </layout>
        </item>
        <item>
         <widget class="QPushButton" name="costBtn">
          <property name="text">
           <string>Cost</string>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QPushButton" name="descentBtn">
          <property name="text">
           <string>Descent</string>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QPushButton" name="accuracyBtn">
          <property name="text">
           <string>Check accuracy</string>
          </property>
         </widget>
        </item>
       </layout>
      </widget>
     </item>
     <item>
      <spacer name="verticalSpacer">
       <property name="orientation">
        <enum>Qt::Vertical</enum>
       </property>
       <property name="sizeHint" stdset="0">
        <size>
         <width>20</width>
         <height>40</height>
        </size>
       </property>
      </spacer>
     </item>
    </layout>
   </item>
   <item>
    <widget class="QScrollArea" name="scrollArea">
     <property name="widgetResizable">
      <bool>true</bool>
     </property>
     <widget class="Perceptron" name="perceptron">
      <property name="geometry">
       <rect>
        <x>0</x>
        <y>0</y>
        <width>737</width>
        <height>602</height>
       </rect>
      </property>
     </widget>
    </widget>
   </item>
  </layout>
 </widget>
 <customwidgets>
  <customwidget>
   <class>ReadMNIST</class>
   <extends>QWidget</extends>
   <header location="global">readmnist.h</header>
   <container>1</container>
  </customwidget>
  <customwidget>
   <class>Perceptron</class>
   <extends>QWidget</extends>
   <header location="global">perceptron.h</header>
   <container>1</container>
   <slots>
    <slot>saveAll()</slot>
   </slots>
  </customwidget>
 </customwidgets>
 <resources/>
 <connections/>
</ui>
